ext/pdo_dblib/dblib_driver.c

```diff
@@ -160,6 +160,29 @@ static zend_string* dblib_handle_quoter(pdo_dbh_t *dbh, const zend_string *unquo
 	if ((paramtype & PDO_PARAM_STR_CHAR) == PDO_PARAM_STR_CHAR) {
 		use_national_character_set = 0;
 	}
+	/*
+	 * A user could be passing a binary (i.e. an image file) in a query.
+	 * It's fragile trying to escape it as a string, so encode it as a
+	 * binary literal instead.
+	 */
+	if (paramtype == PDO_PARAM_BINARY) {
+		is_binary = true;
+	}
+
+	if (is_binary) {
```

Comment:
Why is there a separate is_binary variable instead, I'd inline the `paramtype == PDO_PARAM_BINARY` check directly here.